sqlglot/parser.py

```diff
@@ -2001,7 +2001,9 @@ def _parse_primary(self):
             self._match_r_paren()
 
             if isinstance(this, exp.Subqueryable):
-                this = self._parse_set_operations(self._parse_subquery(this))
+                this = self._parse_set_operations(
+                    self._parse_subquery(this=this, parse_alias=False)
```

Comment:
can you write a test case for this? what's going on here

Reply:
I was also confused, but I think there's a test in this PR that covers this:

```python
SELECT CAST((SELECT x FROM y) AS VARCHAR) AS test
```

Either way, it seems non-trivial, so maybe a comment could be used here?

Reply:
i got it now